tests/unit/test_handlers.py

```diff
@@ -332,6 +332,50 @@ def test_get_template_has_error_response(self):
         # an error response.
         self.assertEqual(original, handler_input)
 
+    def test_encode_json_policy(self):
+        params = {
+            'Document': {
+                'foo': 'bar'
+            },
+            'Other': 'baz'
+        }
+        service_def = {
```

Comment:
I can see where it might be simpler in some areas, but it doesn't look like it will be all that much help in this case since I'm building a ServiceModel, not just a StructureShape. FWIW I'm basing this test on the test immediately below.